pkg/pillar/dpcmanager/dpcmanager.go

```diff
@@ -457,7 +458,9 @@ func (m *DpcManager) GetDNS() types.DeviceNetworkStatus {
 }
 
 func (m *DpcManager) updateGCP(ctx context.Context, gcp types.ConfigItemValueMap) {
```

Comment:
AFAICT this function ends up being called at the top of nim.run(). Doesn't seem like we have everything in place then such as a generated last resort and haven't done the ingestDevicePortConfig() either. Am I missing something?

Reply:
This is inside DpcManager, which is a separate component (and Go routine) used by nim.
Here the [persisted DPCs are first ingested](https://github.com/milan-zededa/eve/blob/dpc-verify-fix/pkg/pillar/dpcmanager/dpcmanager.go#L230-L231), and then it [enters a main even loop ](https://github.com/milan-zededa/eve/blob/dpc-verify-fix/pkg/pillar/dpcmanager/dpcmanager.go#L236). `updateGCP` is called inside the loop [here](https://github.com/milan-zededa/eve/blob/dpc-verify-fix/pkg/pillar/dpcmanager/dpcmanager.go#L247).
So to summarize, DpcManager first loads persisted DPCs and only then starts processing messages from nim.

Reply:
I should also point out that DpcManager is started by nim [here](https://github.com/milan-zededa/eve/blob/dpc-verify-fix/pkg/pillar/cmd/nim/nim.go#L191-L194), which is before [nim waits for GCP](https://github.com/milan-zededa/eve/blob/dpc-verify-fix/pkg/pillar/cmd/nim/nim.go#L200-L207).

Reply:
Ah my confusion comes from dpcmanager.UpdateGcp and dpcmanager.updateGcp being completely different things; not just the exported vs private/impl version.

In such cases it would make sense to use distinct names such as TriggerFoo then Foo, or handleFoo for the handler.

(In addion I read n.ingestPortConfig as being the ingest of the DPCL but that's on me.)